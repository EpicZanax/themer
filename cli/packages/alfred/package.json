{
  "name": "@themerdev/alfred",
  "version": "2.0.0",
  "description": "An Alfred.app theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/alfred"
  },
  "keywords": [
    "themer",
    "alfred",
    "theme"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/alfred#readme"
}
