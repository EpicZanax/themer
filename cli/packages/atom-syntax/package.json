{
  "name": "@themerdev/atom-syntax",
  "version": "3.0.0",
  "description": "Atom syntax template for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "less": "^4.1.1",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  },
  "repository": {
    "type": "git",
    "url": "git+ssh://git@github.com/themerdev/themer.git",
    "directory": "cli/packages/atom-syntax"
  },
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/atom-syntax#readme",
  "keywords": [
    "themer",
    "atom",
    "syntax",
    "theme"
  ]
}
