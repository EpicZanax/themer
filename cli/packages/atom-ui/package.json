{
  "name": "@themerdev/atom-ui",
  "version": "2.0.0",
  "description": "An Atom UI theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/atom-ui"
  },
  "keywords": [
    "themer",
    "atom",
    "ui",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs",
    "/lib/templates/baseLess.mjs",
    "/lib/templates/buttonsLess.mjs",
    "/lib/templates/editorLess.mjs",
    "/lib/templates/findAndReplaceLess.mjs",
    "/lib/templates/gitLess.mjs",
    "/lib/templates/indexLess.mjs",
    "/lib/templates/keyBindingLess.mjs",
    "/lib/templates/listsLess.mjs",
    "/lib/templates/packageJson.mjs",
    "/lib/templates/panelsLess.mjs",
    "/lib/templates/progressLess.mjs",
    "/lib/templates/sitesLess.mjs",
    "/lib/templates/statusBarLess.mjs",
    "/lib/templates/tabsLess.mjs",
    "/lib/templates/textLess.mjs",
    "/lib/templates/tooltipsLess.mjs",
    "/lib/templates/treeViewLess.mjs",
    "/lib/templates/uiVariablesLess.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/atom-ui#readme",
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "less": "^4.1.1",
    "lodash": "^4.17.4",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
