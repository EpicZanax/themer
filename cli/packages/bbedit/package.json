{
  "name": "@themerdev/bbedit",
  "version": "2.0.0",
  "description": "A BBEdit theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/bbedit"
  },
  "keywords": [
    "themer",
    "bbedit",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/bbedit#readme",
  "dependencies": {
    "color": "^3.0.0",
    "lodash": "^4.17.4",
    "xml": "^1.0.1"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
