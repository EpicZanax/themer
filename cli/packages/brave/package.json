{
  "name": "@themerdev/brave",
  "version": "2.0.0",
  "description": "A Brave theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/brave"
  },
  "keywords": [
    "themer",
    "brave",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/brave#readme",
  "dependencies": {
    "@themerdev/chrome": "^2"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
