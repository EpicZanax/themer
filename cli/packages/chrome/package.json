{
  "name": "@themerdev/chrome",
  "version": "2.0.0",
  "description": "A Chrome theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/chrome"
  },
  "keywords": [
    "themer",
    "chrome",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/chrome#readme",
  "dependencies": {
    "color": "^3.0.0",
    "lodash": "^4.17.4"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
