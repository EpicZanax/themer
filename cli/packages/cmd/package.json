{
  "name": "@themerdev/cmd",
  "version": "2.0.0",
  "description": "A CMD.exe theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/cmd"
  },
  "keywords": [
    "themer",
    "cmd",
    "cmd.exe",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/cmd#readme",
  "dependencies": {
    "lodash-es": "^4.17.21"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^1.0.11",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
