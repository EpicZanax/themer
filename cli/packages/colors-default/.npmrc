tag-version-prefix "@themerdev/colors-default-v"
message "@themerdev/colors-default-v%s"
