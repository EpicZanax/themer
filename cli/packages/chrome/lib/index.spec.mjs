import { dirname } from 'node:path';
import { pickBy } from 'lodash';
import { render, renderInstructions } from './index.mjs';
import { colors } from '@themerdev/colors-default';
import { version } from '../package.json' assert { type: 'json' };
import { describe, expect, it } from 'vitest';

describe('Chrome theme generator', () => {
  it('should render a directory containing a manifest.json', async () => {
    const promisedFiles = render(colors);
    Promise.all(promisedFiles).then((files) => {
      const dirnames = files.map((file) => dirname(file.name));
      expect(dirnames.length).toBe(2);
      expect(dirnames).toContain('Themer Light');
      expect(dirnames).toContain('Themer Dark');
    });
  });
  it('should render properly formatted manifest files', async () => {
    const promisedFiles = render(colors);
    Promise.all(promisedFiles).then((files) => {
      files.forEach((file) => {
        const contents = JSON.parse(file.contents);
        expect(contents.version).toBe(version);
        expect(
          pickBy(contents, (value, key) => key !== 'version'),
        ).toMatchSnapshot();
      });
    });
  });
  it('should provide installation instructions', async () => {
    const files = await Promise.all(render(colors));
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
