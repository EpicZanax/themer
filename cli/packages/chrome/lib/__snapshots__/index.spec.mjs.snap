// Vitest Snapshot v1

exports[`Chrome theme generator > should provide installation instructions 1`] = `
"
1. Launch Chrome and go to \`chrome://extensions\`.
2. Check the \\"Developer mode\\" checkbox at the top.
3. Click the \\"Load unpacked extension...\\" button and choose the desired theme directory (\`Themer Dark\` or \`Themer Light\`).

(To reset or remove the theme, visit \`chrome://settings\` and click \\"Reset to Default\\" in the \\"Appearance\\" section.)
  "
`;

exports[`Chrome theme generator > should render properly formatted manifest files 1`] = `
{
  "manifest_version": 2,
  "name": "Themer Dark",
  "theme": {
    "colors": {
      "bookmark_text": [
        255,
        252,
        255,
      ],
      "frame": [
        40,
        38,
        41,
      ],
      "ntp_background": [
        71,
        66,
        71,
      ],
      "ntp_text": [
        255,
        252,
        255,
      ],
      "omnibox_background": [
        40,
        38,
        41,
      ],
      "omnibox_text": [
        255,
        252,
        255,
      ],
      "tab_background_text": [
        255,
        252,
        255,
      ],
      "tab_background_text_inactive": [
        255,
        252,
        255,
      ],
      "tab_background_text_incognito": [
        255,
        252,
        255,
      ],
      "tab_background_text_incognito_inactive": [
        255,
        252,
        255,
      ],
      "tab_text": [
        255,
        252,
        255,
      ],
      "toolbar": [
        71,
        66,
        71,
      ],
    },
    "tints": {
      "buttons": [
        0.5697167755991286,
        0.5,
        0.5,
      ],
      "frame_inactive": [
        -1,
        -1,
        -1,
      ],
      "frame_incognito": [
        0.7743589743589742,
        1,
        0.55,
      ],
      "frame_incognito_inactive": [
        0.7743589743589742,
        1,
        0.55,
      ],
    },
  },
}
`;

exports[`Chrome theme generator > should render properly formatted manifest files 2`] = `
{
  "manifest_version": 2,
  "name": "Themer Light",
  "theme": {
    "colors": {
      "bookmark_text": [
        40,
        38,
        41,
      ],
      "frame": [
        224,
        220,
        224,
      ],
      "ntp_background": [
        255,
        252,
        255,
      ],
      "ntp_text": [
        40,
        38,
        41,
      ],
      "omnibox_background": [
        224,
        220,
        224,
      ],
      "omnibox_text": [
        40,
        38,
        41,
      ],
      "tab_background_text": [
        40,
        38,
        41,
      ],
      "tab_background_text_inactive": [
        40,
        38,
        41,
      ],
      "tab_background_text_incognito": [
        40,
        38,
        41,
      ],
      "tab_background_text_incognito_inactive": [
        40,
        38,
        41,
      ],
      "tab_text": [
        40,
        38,
        41,
      ],
      "toolbar": [
        255,
        252,
        255,
      ],
    },
    "tints": {
      "buttons": [
        0.5692488262910799,
        0.5,
        0.5,
      ],
      "frame_inactive": [
        -1,
        -1,
        -1,
      ],
      "frame_incognito": [
        0.7745803357314148,
        1,
        0.45,
      ],
      "frame_incognito_inactive": [
        0.7745803357314148,
        1,
        0.45,
      ],
    },
  },
}
`;
